lib/internal/Magento/Framework/Setup/Declaration/Schema/Db/SchemaBuilder.php

```diff
@@ -88,8 +88,22 @@ public function build(Schema $schema)
             foreach ($data['table'] as $keyTable => $tableColumns) {
                 $tableColumns['column'] ??= [];
                 foreach ($tableColumns['column'] as $keyColumn => $columnData) {
-                    if ($columnData['type'] == 'json') {
-                        $tablesWithJsonTypeField[$keyTable] = $keyColumn;
+                    try {
+                        if ($columnData['type'] == 'json') {
+                            $tablesWithJsonTypeField[$keyTable] = $keyColumn;
+                        }
+                    } catch (\Throwable $e) {
+                        // Create a new exception with extended context message
+                        $errorMessage = sprintf(
+                            "%s\nError processing table %s column %s",
+                            $e->getMessage(),
+                            $keyTable,
+                            $keyColumn
+                        );
+                        
+                        // Throw a new exception with the extended message
+                        // This preserves the original error but adds our context
+                        throw new \Exception($errorMessage, $e->getCode(), $e);
```

Comment:
Instead of rethrowing a generic `\Exception`, consider a specific exception type, e.g., `\Magento\Framework\Exception\LocalizedException`

Reply:
unfortunately i can't do this now. However feel free edit this PR.